File: src/main/java/filter/ValidPersonaLogin.java
package filter;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Target({ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER,
        ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = PersonaLoginValidator.class)
@Documented
public @interface ValidPersonaLogin {
    String message () default "Hubo un error";

    Class<?>[] groups () default {};

    Class<? extends Payload>[] payload () default {};
}

File: src/main/java/filter/ValidPersonaLogin.kt
package filter

import javax.validation.Constraint
import javax.validation.Payload
import java.lang.annotation.*
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.CLASS, AnnotationTarget.FILE)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [PersonaLoginValidator::class])
@Documented
annotation class ValidPersonaLogin(val message: String = "Hubo un error", val groups: Array<KClass<*>> = [], val payload: Array<KClass<out Payload>> = [])

